src/converters/app-converter.ts

```diff
@@ -169,6 +164,72 @@ export class StackConverter extends ArtifactConverter {
         }
     }
 
+    /**
+     * This function takes a Pulumi resource that was mapped from a CFN resource and
+     * "registers" it to the StackConverter. The StackConverter keeps track of all of the
+     * resources that were mapped from the CFN resources in the stack. It uses this information
+     * to resolve references to other resources in the stack.
+     *
+     * Typically there will be a couple of scenarios for how a CFN resource maps to Pulumi resources:
+     *
+     * 1. The CFN resource maps to a single Pulumi aws-native resource
+     *    This is the most straightforward case because everything maps directly, including the attributes
+     *
+     * 2. The CFN resource maps to single Pulumi aws resources
+     *    In this case the mapping is to a single resource, but some times the attributes available
+     *    on the CFN resource do not map to the attributes on the Pulumi resource. In these cases
+     *    the mapping can return custom attributes that are later available when references are resolved
+     *
+     * 3. The CFN resource maps to multiple Pulumi aws resources
+     *    One example would be the AWS::IAM::Policy resource which in CFN includes Role, Group, and User
+     *    Polices, but in Pulumi aws these are broken out into separate resources. In that case the "main"
+     *    resource would be the Policy and the other resources would be added to the resources array.
+     *    The critical point here is that the "main" resource needs to have a logicalId that matches the
+     *    logicalId of the CFN resource, while the other supporting resources must have different logicalIds.
+     *
+     * @param mapped - The Pulumi Resource that was mapped from the CFN resource
+     * @param node - The GraphNode that represents the CFN Resource
+     */
+    private registerResource(mapped: ResourceMapping, node: GraphNode): void {
+        const cfn = node.resource;
+        // This should always be set because we only call this function when it is, but
```

Comment:
should we just mark it as required in the arg type then? E.g. `Required<Pick<GraphNode, 'resource'>> & GraphNode`?

Reply:
Ah, I had hoped the guard would work. Yeah in that case this is the better approach!